### PR TITLE
[Inductor max autotune] Make autotune_select_algorithm more robust

### PR DESCRIPTION
**Pull request by kadeng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124928

This diff makes sure that a custom exception is thrown when no valid
choices remain during autotuning. This allows to gracefully fall back
to a default choice, even if that default choice has not been passed to
autotune_select_algorithm.

Additionally, this diff handles RuntimeErrors during autotuning gracefully, e.g. the corresponding choice is ignored but it does not lead to the compilation failure of the entire model if a problematic choice is encountered during autotuning.
( An error is being logged, though).

Test Plan:
CI

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by kadeng** [comment]
@int3 I noticed you wrote a diff that's doing something similar. Will take a look and see that this one is still compatible with your changes.

File: torch/_inductor/select_algorithm.py [hunk 1]
```diff
@@ -1115,7 +1117,16 @@ def get_timings():
...
             return choices[0].output_node()
 
-        selected_choice = builtins.min(timings, key=timings.__getitem__).output_node()
+        selected_key = builtins.min(timings, key=timings.__getitem__)
+        selected_time = timings[selected_key]
+        if (
+            (not isinstance(selected_time, float))
+            or (selected_time < 0.0)
+            or (not math.isfinite(selected_time))
+            or math.isnan(selected_time)
```
Comment (int3) [inline torch/_inductor/select_algorithm.py:1126]: feels a little overkill (pretty sure we only generate inf and regular floats) but not a big deal

**Comment by int3** [comment]
Ah I didn't see you had this in the works. Yeah, this is more or less compatible with my changes. I can add my tests after you land this.

**Review by int3 — APPROVED** [review]
lgtm

**Pull request closed without merging**